test/metabase/task/setup/query_analysis_setup.clj

```diff
@@ -36,31 +33,17 @@
                               :query_type    "native"
                               :dataset_query (mt/native-query {:query "SELECT id FROM venues"})}]
 
-      ;; Make sure there is no existing analysis for the relevant cards
-      (t2/delete! :model/QueryField :card_id [:in (map :id [c1 c2 c3 c4 arch])])
-
-      ;; Make sure some other card has analysis
-      (testing "There is at least one card with existing analysis"
-        (query-analysis/analyze-card! (:id c3))
-        (is (pos? (count (t2/select :model/QueryField :card_id (:id c3))))))
-
-      (let [queued-ids   (atom #{})
-            expected-ids (into #{} (map :id) [c1 c2 c4])]
-
-        ;; Run the backfill with a mocked out publisher
-        (#'backfill/backfill-missing-query-fields!
-         #(swap! queued-ids conj (:id %)))
-
-        (testing "The expected cards were all sent to the analyzer"
-          (is (= expected-ids (set/intersection expected-ids @queued-ids))))
+                  ;; Make sure there is no existing analysis for the relevant cards
+                  (t2/delete! :model/QueryField :card_id [:in (map :id [c1 c2 c3 c4 arch])])
 
-        (testing "The card with existing analysis was not sent to the analyzer again"
-          (is (not (@queued-ids (:id c3)))))))))
+                  ;; Make sure some other card has analysis
+                  (query-analysis/analyze-card! (:id c3))
 
+                  (mt/call-with-map-params f [c1 c2 c3 c4 arch]))))
 
-(comment
-  (set! *warn-on-reflection* true)
-  (queue/clear! @#'query-analysis/queue)
-  (.-queued-set @#'query-analysis/queue)
-  (.peek (.-async-queue @#'query-analysis/queue))
-  (.peek (.-sync-queue @#'query-analysis/queue)))
+(defmacro with-test-setup!
+  "Set up the data required to test the Query Analysis related tasks"
+  [& body]
+  `(do-with-test-setup!
+    (mt/with-anaphora [c1 c2 c3 c4 arch]
+                      ~@body)))
```

Comment:
ah whoops

https://github.com/metabase/metabase/pull/45716

Reply:
I was shamefully using IntelliJ, so I'll also need to fix it in my um settings.jar (snazzy)